tests/harvester/test_source_harvester.py

```diff
@@ -1,30 +1,43 @@
-from pathlib import Path
+from pytest_mock import MockerFixture
 
-import pytest
+import pandas as pd
 
 from dlme_airflow.harvester.source_harvester import data_source_harvester
+from dlme_airflow.models.collection import Collection
+from dlme_airflow.models.provider import Provider
 
 
-@pytest.fixture
-def csv_fixture(monkeypatch):
-    def mockglob(*args):
-        return [{"title": "A title", "url": "https://arabic.io/"}]
+class MockTaskInstance:
+    def __init__(self):
+        self.xcom_dict = {}
 
-    monkeypatch.setattr(Path, "glob", mockglob)
+    def xcom_push(self, key: str, value):
+        self.xcom_dict[key] = value
 
 
-@pytest.mark.skip(reason="TODO Fix tests with mocked provider collection.")
 def test_source_harvester():
     assert data_source_harvester
 
 
-@pytest.mark.skip(reason="TODO Fix tests with mocked provider collection.")
-def test_provider_not_found():
-    with pytest.raises(AttributeError, match=r"bad_provider"):
-        data_source_harvester(provider="bad_provider")
+def test_dataframe_retrieval_and_profiling(mocker: MockerFixture):
+    collection = Collection(Provider("aims"), "aims")
+    mock_task_instance = MockTaskInstance()
 
+    column_headers = ["id", "source"]
+    rows = [
+        column_headers,
+        ["1", ["Rare Books and Special Collections Library"]],
+        ["2", ["Rare Books and Special Collections Library"]],
+    ]
```

Comment:
i wasn't sure if building my mock dataframe this way actually realistically simulated the dataframe we get from intake, but this did give me an extra header row, and the dataframe looked reasonable otherwise (caveat: i'm very new to pandas and dataframes).

happy to re-work if there's a better way to setup this test.